src/support/small_vector.h

```diff
@@ -161,7 +165,14 @@ template<typename T, size_t N> class SmallVector {
       return index != other.index || parent != other.parent;
     }
 
-    void operator++() { index++; }
+    Iterator& operator++() {
+      index++;
+      return *this;
+    }
+    Iterator operator++(int) {
+      index++;
+      return *this;
+    }
```

Comment:
We should at least make this a correct postincrement implementation.
```suggestion
    Iterator operator++(int) {
      Iterator self = *static_cast<Iterator*>(this);
      index++;
      return self;
    }
```